test/services/database-analyzer/database-analyzer-mongo.test.js

```diff
@@ -31,22 +32,27 @@ const expectedSubDocumentsAmbiguousIds = require('../../../test-expected/mongo/d
 const expectedSubDocumentsUsingIds = require('../../../test-expected/mongo/db-analysis-output/sub-documents-using-ids.expected');
 const expectedSubDocumentsNotUsingIds = require('../../../test-expected/mongo/db-analysis-output/sub-documents-not-using-ids.expected');
 const expectedSubDocumentUsingIds = require('../../../test-expected/mongo/db-analysis-output/sub-document-using-ids.expected');
+const expectedComplexModelWithAView = require('../../../test-expected/mongo/db-analysis-output/complex-model-with-a-view.expected');
 
 function getMongoHelper(mongoUrl) {
   return new MongoHelper(mongoUrl);
 }
 
-async function getAnalyzerOutputWithModel(mongoUrl, model) {
+async function getAnalyzerOutput(mongoUrl, callback) {
   const mongoHelper = await getMongoHelper(mongoUrl);
   const databaseConnection = await mongoHelper.connect();
   await mongoHelper.dropAllCollections();
-  await mongoHelper.given(model);
+  await callback(mongoHelper);
```

Comment:
I thought it would be more scalable to have this:

```
getAnalyzerOutputWithModel
getAnalyzerOutput
```

Over this:

```
getAnalyzerOutputWithModel // with an optional param callback or not
// Caveat: one more option in function, and a if, which increase cognitive complexity.
```

But it's an **opinion**, so I can change if you prefer.